app/src/main/java/com/sunasterisk/moviedb_51/utils/BindingUtils.kt

```diff
@@ -123,3 +125,46 @@ fun ChipGroup.setItemGenreDetails(genres: List<Genres>?) {
         addView(genresChip)
     }
 }
+
+@BindingAdapter("setMarginTop")
+fun AppBarLayout.setMarginTop(margin: Int?) {
+    margin ?: return
+    val params = this.layoutParams as ViewGroup.MarginLayoutParams
+    params.topMargin = margin
+    this.layoutParams = params
+}
+
+@BindingAdapter("itemsGenreMovies")
+fun ChipGroup.setItemsGenreMovies(genreIds: List<Int>?) {
+    genreIds ?: return
+    if (childCount > 0) removeAllViews()
+    for (i in genreIds.indices) {
+        if (i < COUNT_CHIP_MOVIES) {
+            val genresChip = LayoutInflater.from(context)
+                .inflate(R.layout.item_chip_details, this, false) as Chip
+            genresChip.text = when (genreIds[i]) {
+                GenresTypes.ACTION.genreId -> context.getString(GenresTypes.ACTION.genreNameId)
+                GenresTypes.ADVENTURE.genreId -> context.getString(GenresTypes.ADVENTURE.genreNameId)
+                GenresTypes.ANIMATION.genreId -> context.getString(GenresTypes.ANIMATION.genreNameId)
+                GenresTypes.COMEDY.genreId -> context.getString(GenresTypes.COMEDY.genreNameId)
+                GenresTypes.CRIME.genreId -> context.getString(GenresTypes.CRIME.genreNameId)
+                GenresTypes.DOCUMENTARY.genreId -> context.getString(GenresTypes.DOCUMENTARY.genreNameId)
+                GenresTypes.DRAMA.genreId -> context.getString(GenresTypes.DRAMA.genreNameId)
+                GenresTypes.FAMILY.genreId -> context.getString(GenresTypes.FAMILY.genreNameId)
+                GenresTypes.FANTASY.genreId -> context.getString(GenresTypes.FANTASY.genreNameId)
+                GenresTypes.HISTORY.genreId -> context.getString(GenresTypes.HISTORY.genreNameId)
+                GenresTypes.HORROR.genreId -> context.getString(GenresTypes.HORROR.genreNameId)
+                GenresTypes.MUSIC.genreId -> context.getString(GenresTypes.MUSIC.genreNameId)
+                GenresTypes.MYSTERY.genreId -> context.getString(GenresTypes.MYSTERY.genreNameId)
+                GenresTypes.ROMANCE.genreId -> context.getString(GenresTypes.ROMANCE.genreNameId)
+                GenresTypes.SCIENCE_FICTION.genreId -> context.getString(GenresTypes.SCIENCE_FICTION.genreNameId)
+                GenresTypes.TV_MOVIE.genreId -> context.getString(GenresTypes.TV_MOVIE.genreNameId)
+                GenresTypes.THRILLER.genreId -> context.getString(GenresTypes.THRILLER.genreNameId)
+                GenresTypes.WAR.genreId -> context.getString(GenresTypes.WAR.genreNameId)
+                GenresTypes.WESTERN.genreId -> context.getString(GenresTypes.WESTERN.genreNameId)
+                else -> context.getString(GenresTypes.NONE.genreNameId)
```

Comment:
trường hợp này a thấy dùng for có vẻ đúng đắng hơn. Bị lặp code 2 lần và quá dài dòng